common/src/main/java/com/mineaurion/aurionchat/common/command/ChatCommandCommon.java

```diff
@@ -90,6 +91,10 @@ public void defaultCommand(AurionChatPlayer aurionChatPlayer){
 
     public boolean execute(AurionChatPlayer aurionChatPlayer, String channel, Action action) {
         try {
+            String perm = "aurionchat.channel." + action.name().toLowerCase() + "." + channel;

```

Comment:
I think it will be better to rewrite the function `checkChannelExist` and launch this function directly here and remove the usage in the "sub function" leave/join/spy to directly if the channel exist and the player permission

In this project i don't centralize even futher the command in the common so there is a little bit of work in forge and fabric since the command suggest all the channel (tab completion)

Reply:
Yes, consistent tab completion was not really in my priorities here for the moment.
I'm afraid I'm not quite sure what you mean though; the `checkChannelExists` call always precedes this code; so its never skipped nor is there false-positive error messages. Am I missing something?